modules/cudafeatures2d/test/test_features2d.cpp

```diff
@@ -102,6 +108,62 @@ CUDA_TEST_P(FAST, Accuracy)
     }
 }
 
+#ifdef _OPENMP
+CUDA_TEST_P(FAST, Async)
+{
+    cv::Mat image_ = readImage("features2d/aloe.png", cv::IMREAD_GRAYSCALE);
+    ASSERT_FALSE(image_.empty());
+
+    if (!supportFeature(devInfo, cv::cuda::GLOBAL_ATOMICS))
+    {
+        try
+        {
```

Comment:
Do we really need this code path?

Perhaps it is better to skip test:
```
if (!supportFeature(devInfo, cv::cuda::GLOBAL_ATOMICS))
    throw SkipTestException("CUDA device doesn't support global atomics");
```

Reply:
I think you are right.